### PR TITLE
[T3A3][W11-A4] Cherlynn Cha

### PR DESCRIPTION
**Pull request by Cherc**

### CODE REVIEW

**Comment by Cherc** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -14,6 +14,10 @@
 
     public final String value;
     private boolean isPrivate;
+    public final Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: think why you don't / shouldn't make it public here

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -26,6 +30,17 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
         this.value = address;
+        String[] parts = address.split(", ");
+        
+        block = new Block();
+        street = new Street();
+        unit = new Unit();
+        postal = new PostalCode();
+        
+        this.block.setBlock(parts[0]);
+        this.street.setStreet(parts[1]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: can you provide this information in the constructor, why? why not?

File: src/seedu/addressbook/data/person/Block.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: please add some header comments for the class to tell what it is supposed to do.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -52,14 +52,18 @@ public static boolean isValidAddress(String test) {
...
-        return value;
+        return block.getBlock() + street.getStreet() + unit.getUnit() + postal.getPostal();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.block.getBlock().equals( ((Address) other).block.getBlock() )
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: you can think of implementing a (or an!) equals method  in each class and use that here.

similarly the hashcode.

File: doc/UserGuide.md [hunk 5]
```diff
@@ -34,7 +34,7 @@ Format: `help`
  
 ### Adding a person: `add`
 Adds a person to the address book<br>
-Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/BLOCK, STREET, UNIT, POSTALCODE [t/TAG]...`
```
Comment (okkhoy) [inline doc/UserGuide.md]: good job here, updating the user guide :) :+1:

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -26,6 +30,17 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: it doesn't seem like you updated the help? please take note that you need to update the help, tests and user guide when you make such modifications.

**Comment by okkhoy** [comment]
@Cherc Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**